torch/distributed/checkpoint/planner_helpers.py

```diff
@@ -197,8 +227,39 @@ def _create_write_items(fqn: str, object: Any) -> List[WriteItem]:
         return [_create_write_item_for_bytesio(fqn, object)]
 
 
+def _create_shard_from_dtensor(tensor: DTensor) -> Shard:
+    device_mesh = tensor.device_mesh
+    assert (
+        device_mesh.ndim == 1
+    ), "Only 1D DeviceMeshes can currently be handled."
+
+    sizes = tensor._spec.local_shape
+    offsets = tensor._spec.local_offsets
+    return Shard(
+        tensor=tensor.to_local(),
+        metadata=ShardMetadata(
+            shard_offsets=list(offsets),
+            shard_sizes=list(sizes),
+            placement=f"rank:{dist.get_rank()}/{tensor.to_local().device}",
+        ),
+    )
+
+
 def _create_read_items(fqn: str, md: STORAGE_TYPES, obj: Any) -> List[ReadItem]:
-    if isinstance(md, BytesStorageMetadata):
+    if not isinstance(md, BytesStorageMetadata):
```

Comment:
It seems that the original `if-else` structure is cleaner -- no nested if is required.

Reply:
> It seems that the original `if-else` structure is cleaner -- no nested if is required.

Ye. I also think 'if-else' is cleaner, but mypy would raise a static typing error since it would think the metadata in _create_sharded_read_items(fqn, md, local_shards) could either be TensorStorageMetadata or BytesStorageMetadata while it could only be TensorStorageMetadata. To fix the mypy error, I have to move it around like this.